tests/src/main/java/com/hazelcast/stabilizer/tests/map/PredicatesTest.java

```diff
@@ -0,0 +1,351 @@
+            EmployeeInterface value = createEmployee(id);
+            map.put(key, value);
+        }
+        log.info("Map size is:" + map.size());
+    }
+
+    @Run
+    public void run() {
+        ThreadSpawner spawner = new ThreadSpawner(testContext.getTestId());
+        for (int k = 0; k < threadCount; k++) {
+            spawner.spawn(new Worker());
+        }
+        spawner.awaitCompletion();
+    }
+
+    @Performance
+    public long getOperationCount() {
+        return operations.get();
+    }
+
+    private class Worker implements Runnable {
+        @Override
+        public void run() {
+            long iteration = 0;
+            Metronome metronome = SimpleMetronome.withFixedIntervalMs(intervalMs);
+            SqlPredicate sqlPredicate = new SqlPredicate(sqlQuery);
+            NamePredicate namePredicate = new NamePredicate("aaa");
+            while (!testContext.isStopped()) {
+                metronome.waitForNext();
+                search.started();
+                if (customPredicate) {
+                    map.values(namePredicate);
+                } else {
+                    map.values(sqlPredicate);
+                }
+                search.done();
+
+                iteration++;
+                if (iteration % logFrequency == 0) {
+                    log.info(Thread.currentThread().getName() + " At iteration: " + iteration);
+                }
+
+                if (iteration % performanceUpdateFrequency == 0) {
+                    operations.addAndGet(performanceUpdateFrequency);
+                }
+
+            }
+            operations.addAndGet(iteration % performanceUpdateFrequency);
+        }
+    }
+
+    public EmployeeInterface createEmployee(int id) {
+        switch (serializationStrategy) {
+            case DATA_SERIALIZABLE:
+                return new DataSerializableEmployee(id);
+            case JAVA_SERIALIZABLE:
+                return new SerializableEmployee(id);
+            case IDENTIFIED_DATA_SERIALIZABLE:
+                return new IdentifiedDataSerializableEmployee(id);
+            case PORTABLE:
+                return new PortableEmployee(id);
+        }
+        return null;
+    }
+
+    //Custom Predicates
+    private static class NamePredicate implements Predicate<String, EmployeeInterface> {
+        private final String name;
+
+        private NamePredicate(String name) {
+            this.name = name;
+        }
+
+        @Override
+        public boolean apply(Map.Entry<String, EmployeeInterface> entry) {
+            EmployeeInterface value = entry.getValue();
+            return value.getName().equals(name);
+        }
+    }
+
+    public interface EmployeeInterface {
+
+        public String getName();
+    }
+
+    public static class DataSerializableEmployee implements DataSerializable,EmployeeInterface {
+
+        protected int id;
+        protected String name;
+
+        public String getName() {
```

Comment:
I'm fine with the repeated methods. The problem here is the Java serialization that doesn't deal well with super classes that are not implementing Serializable. And therefor you need to put the fields in an object that does implement Serializable.